internal/provider/common.go

```diff
@@ -175,3 +184,24 @@ func buildIPSetForTenableFormat(iplist string) (*netaddr.IPSet, error) {
 
 	return builder.IPSet()
 }
+
+func validateRecastAcceptRiskProtocol(protocol any, path cty.Path) (diags diag.Diagnostics) {
+
+	if _, err := getRecastAcceptRiskProtocolID(protocol.(string)); err != nil {
+		diags = append(diags, diag.Diagnostic{
+			Severity:      diag.Error,
+			Summary:       fmt.Sprintf("failed to get protocol id: %s", err),
+			AttributePath: path,
+		})
+	}
+	return
+}
+
+func getRecastAcceptRiskProtocolID(protocol string) (string, error) {
+	id, ok := RecastAcceptRiskProtocolIDMap[protocol]
```

Comment:
In the spirit of being generous in what we accept, can we accept arbitrary casings? I'd be unsurprised if people were tripped by wanting to use 'tcp' instead of 'TCP' and _amazed_ if noone tripped on 'Unknown' requiring capitalization.